rufff is not needed to run tests.
We de not have ruff packaged (yet).

--- setuptools_scm-9.2.1/pyproject.toml.orig
+++ setuptools_scm-9.2.1/pyproject.toml
@@ -68,7 +68,6 @@
   "pytest",
   "pytest-timeout",  # Timeout protection for CI/CD
   "rich",
-  "ruff",
   "mypy~=1.13.0", # pinned to old for python 3.8
   'typing-extensions; python_version < "3.11"',
   "wheel",
